src/resource/filters.js

```diff
@@ -194,15 +198,38 @@ export default function filtersFactory($container, config) {
             this.render($container);
         })
         .on('render', function() {
-            if (this.config.data) {
-                this.update(this.config.data);
-            }
+            Promise.race([
+                new Promise(resolve => {
+                    if (this.config.data) {
+                        this
```

Comment:
I'm wondering if the  `update` method shouldn't return a Promise (in addition to the event) this would make the workflow more straight forward ?

Reply:
Yes that would be better. That will also probably requires some rework on the consumer. I will try that approach.